sale_coupon_criteria_multi_product/models/sale_coupon_criteria.py

```diff
@@ -0,0 +1,40 @@
+# Copyright 2021 Tecnativa - David Vidal
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class SaleCouponCriteria(models.Model):
+    _name = "sale.coupon.criteria"
+    _description = "Coupon Multi Product Criteria"
+
+    program_id = fields.Many2one(comodel_name="sale.coupon.program",)
+    rule_min_quantity = fields.Integer(
+        string="Minimum Quantity",
+        compute="_compute_rule_min_quantity",
+        store=True,
+        readonly=False,
+        help="Minimum required product quantity to get the reward",
+    )
+    product_ids = fields.Many2many(comodel_name="product.product", string="Products",)
```

Comment:
Does `required` work on m2m?

```suggestion
    product_ids = fields.Many2many(comodel_name="product.product", string="Products", required=True)
```

Reply:
Somehow...:

![m2m-required](https://user-images.githubusercontent.com/5040182/126457133-02a8b57c-0eab-47ba-afe6-b4fc16f2693a.gif)